root_backend_checks.tf

```diff
@@ -24,3 +24,247 @@ module "backend_checks_api" {
+    }
+  ]
+}
+
+module "checksum_v2" {
+  source        = "./tdr-terraform-modules/generic_lambda"
+  tags          = local.common_tags
+  function_name = local.checksum_v2_function_name
+  handler       = "uk.gov.nationalarchives.checksum.Lambda::process"
+  policies = {
+    "TDRChecksumV2LambdaPolicy${title(local.environment)}" = templatefile("./templates/iam_policy/lambda_s3_only_policy.json.tpl", {
+      function_name = local.checksum_v2_function_name,
+      account_id    = data.aws_caller_identity.current.account_id,
+      bucket_name   = local.upload_files_cloudfront_dirty_bucket_name
+    })
+  }
+  role_name = "TDRChecksumV2LambdaRole${title(local.environment)}"
+  runtime   = "java11"
+  plaintext_env_vars = {
+    CHUNK_SIZE_IN_MB = 50
+    S3_BUCKET        = local.upload_files_cloudfront_dirty_bucket_name
+  }
+  vpc_config = [
+    {
+      subnet_ids         = module.shared_vpc.private_subnets
+      security_group_ids = [module.outbound_only_security_group.security_group_id]
+    }
+  ]
+}
+
+module "redacted_files" {
+  source        = "./tdr-terraform-modules/generic_lambda"
+  tags          = local.common_tags
+  function_name = local.redacted_files_function_name
+  handler       = "uk.gov.nationalarchives.Lambda::run"
+  policies      = {}
+  role_name     = "TDRRedactedFilesLambdaRole${title(local.environment)}"
+  runtime       = "java11"
+  vpc_config = [
+    {
+      subnet_ids         = module.shared_vpc.private_subnets
+      security_group_ids = [module.no_access_security_group.security_group_id]
+    }
+  ]
+}
+
+module "statuses" {
+  source        = "./tdr-terraform-modules/generic_lambda"
+  tags          = local.common_tags
+  function_name = local.statuses_function_name
+  handler       = "uk.gov.nationalarchives.Lambda::run"
+  policies = {
+    "TDRStatusesLambdaPolicy${title(local.environment)}" = templatefile("./templates/iam_policy/allow_iam_db_auth.json.tpl", {
+      function_name = local.statuses_function_name,
+      account_id    = data.aws_caller_identity.current.account_id,
+      resource_id   = module.consignment_api_database.resource_id
+      user_name     = local.consignment_user_name
+    })
+  }
+  role_name = "TDRStatusesLambdaRole${title(local.environment)}"
+  runtime   = "java11"
+  plaintext_env_vars = {
+    USER_NAME    = local.consignment_user_name
+    URL_PATH     = local.url_path
+    USE_IAM_AUTH = true
+  }
+  vpc_config = [
+    {
+      subnet_ids         = module.shared_vpc.private_subnets
+      security_group_ids = [module.outbound_with_db_security_group.security_group_id]
+    }
+  ]
+}
+
+module "yara_av_v2" {
+  source        = "./tdr-terraform-modules/generic_lambda"
+  tags          = local.common_tags
+  function_name = local.yara_av_v2_function_name
+  handler       = "matcher.matcher_lambda_handler"
+  policies = {
+    "TDRChecksumV2LambdaPolicy${title(local.environment)}" = templatefile("./templates/iam_policy/lambda_s3_only_policy.json.tpl", {
```

Comment:
Is that the correct policy name, should it not be `TDRYaraAVV2LambdaPolicy...`?

Reply:
Good spot. I changed it, thanks.